src/sandstorm/grain.capnp

```diff
@@ -23,80 +23,165 @@ using Util = import "util.capnp";
-# is much like Android intents.)
-
-interface PowerboxCapability {
-  # Capabilities to be offered to the powerbox must implement PowerboxCapability (in addition to
-  # the interface for the application functionality they provide).  PowerboxCapability provides
-  # metadata about the capability for display in the powerbox UI as well as the sharing graph
-  # (which shows inter-app capabilities).
-  #
-  # Capabilities sent directly between two grains (e.g. as an RPC parameter) rather than through
-  # the powerbox UI may also wish to implement PowerboxCapability in order to improve the sharing
-  # graph visualization.  Only persisted capabilities will ever show up in the visualization.
-
-  getPowerboxInfo @0 () -> PowerboxInfo;
-  struct PowerboxInfo {
-    title @0 :Util.LocalizedText;
-    # Title for this capability if displayed as an option in the powerbox.  Useful for
-    # distinguishing from other capabilities exported by the same grain.  Leave blank if the
-    # capability effectively represents the entire grain and so should just take the grain's title.
+# In "request" mode, an app initiates the powerbox by requesting to receive a capability matching
+# some particluar criteria using `SessionContext.request()` (or though the client-side
+# postMessage() API, described in the documentation for `SessionContext.request()`). The user is
+# presented with a list of other grains of theirs which might be able to fulfill this request and
+# asked to choose one. Other grains initially register their ability to answer certain requests
+# by filling in the powerbox fields of `UiView.ViewInfo`. When the user chooses a grain,
+# `UiView.newRequestSession()` is called on the providing grain and the resulting UI session is
+# displayed embedded in the powerbox. The providing grain can render a UI which prompts the user
+# for additional details if needed, or implements some sort of additional picker. Once the grain
+# knows which capability to provide, it calls `SessionContext.provide()` to fulfill the original
+# request.
+#
+# In "offer" mode, an app initiates the powerbox by calling `SessionContext.offer()` in a normal,
+# non-powerbox session, to indicate that it wishes to offer some capability to the current user
+# for use in other apps. The user is presented with a list of apps and grains that are able to
+# accept this offer. Grains can register interest in receiving offers by filling in the powerbox
+# metadata in `UiView.ViewInfo`. Apps can also indicate in their manifest that it makes sense for a
+# user to create a whole new grain to accept a powerbox offer. In either case, a session is created
+# using `UiView.newOfferSession()`.
+
+struct PowerboxDescriptor {
+  # Describes properties of capabilities exported by the powerbox, or capabilities requested
+  # through the powerbox.
+  #
+  # Usually, descriptors come in a list, where the list indicates one of two things:
```

Comment:
Oops, I was mistaken here. `List(PowerboxDesciptor)` is indeed typical, and specifies a disjunction of descriptors, where each descriptor is a conjunction.